WooCommerce/WooCommerceTests/ViewRelated/Dashboard/Settings/In-Person Payments/Onboarding Errors/InPersonPaymentsCashOnDeliveryPaymentGatewayNotSetUpViewModelTests.swift

```diff
@@ -0,0 +1,105 @@
+import XCTest
+import TestKit
+
+@testable import WooCommerce
+import Yosemite
+import Networking
+
+final class InPersonPaymentsCashOnDeliveryPaymentGatewayNotSetUpViewModelTests: XCTestCase {
+
+    private var stores: MockStoresManager!
+
+    private var noticePresenter: MockNoticePresenter!
+
+    private var sut: InPersonPaymentsCashOnDeliveryPaymentGatewayNotSetUpViewModel!
+
+    override func setUp() {
+        stores = MockStoresManager(sessionManager: .testingInstance)
```

Comment:
Should we use `makeForTesting()` here rather than `.testingInstance` to avoid potentially using the same instance of MockStoresManager for different tests? Ref:  https://github.com/woocommerce/woocommerce-ios/issues/2885 . That said I'm not sure if this makes any difference as is declared inside the `setUp()` method and should be teared down each time

Reply:
It doesn't actually make a difference. `testingInstance` is a static computed var which gets called for each test, and in turn calls `SessionManager.init`, and returns the result... so it's a new one every time.

You can confirm this with a breakpoint in `testingInstance`, and checking the output of `po sessionManager` for each time it's hit when running this test class – here's mine:

```
(lldb) po sessionManager
<SessionManager: 0x600001302f00>
(lldb) po sessionManager
<SessionManager: 0x600001337a00>
(lldb) po sessionManager
<SessionManager: 0x600001308e00>
(lldb) po sessionManager
<SessionManager: 0x60000130c800>
```

So I think the bug you reference is incorrect. I'll pop a comment on and close it.

However, I am going to make this change. This is because `testingInstance` is _named_ like a singleton, even though it isn't one, so this code does imply that we're using the same `sessionManager` for each test, and so it would be slightly clearer to use `makeForTesting()`.